projects/plugins/jetpack/modules/subscriptions/jetpack-user-content-link-redirection.php

```diff
@@ -48,6 +69,29 @@ function jetpack_user_content_link_redirection() {
 	exit;
 }
 
+/**
+ * Check if the domain of the request is the same as the domain of the given blog_id
+ *
+ * @param string $domain_from_request The domain from the request.
+ * @param int    $blog_id             The blog_id to check against.
+ *
+ * @return bool
+ */
+function jetpack_user_content_link_is_same_domain( $domain_from_request, $blog_id ) {
+	$request  = sprintf( '/sites/%d?fields=URL', $blog_id );
+	$response = Client::wpcom_json_api_request_as_blog( $request );
```

Comment:
I'm not too sure about calling home everytime a link is accessed, especially just to get the site URL. Is it not possible to get the site URL locally? If not, some caching would be nice.

Reply:
Indeed, the ideal would be to get it locally. I did some research and couldn't find a way to do that. I'll try a bit more tomorrow, but if I can't find anything on the local side, including a cache would be a good alternative.

Reply:
Moved to the blog_id check instead to simplify the implementation.